internals/cli/cli.go

```diff
@@ -80,9 +80,9 @@ var commands []*cmdInfo
 // debugCommands holds information about all debug commands.
 var debugCommands []*cmdInfo
 
-// addCommand replaces parser.addCommand() in a way that is compatible with
+// AddCommand replaces parser.AddCommand() in a way that is compatible with
 // re-constructing a pristine parser.
-func addCommand(name, shortHelp, longHelp string, builder func() flags.Commander, optDescs map[string]string, argDescs []argDesc) *cmdInfo {
+func AddCommand(name, shortHelp, longHelp string, builder func() flags.Commander, optDescs map[string]string, argDescs []argDesc) *cmdInfo {
```

Comment:
Now that this is exported, maybe we can reconsider the signature. It has 6 args, which is a bit unwieldy. What about exporting a `CommandInfo` struct so these can become "named args" (or even just exporting the relevant fields from `CmdInfo`).

```
func AddCommand(info *CommandInfo) *cmdInfo { ... }

...

cli.AddCommand(&cli.CommandInfo{
    Name: "foo",
    ShortHelp: "short help",
    ...
})
```

Also, does it need to return anything? Only cmd_enter and cmd_exec use the result value to add "extra", but maybe they can still call the internal one? Or we expose `Extra` in the passed-in struct too?